src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -215,9 +215,12 @@ internal void AddProject(AbstractProject project)
 
         internal void StartPushingToWorkspaceAndNotifyOfOpenDocuments(IEnumerable<AbstractProject> projects)
         {
-            foreach (var hostState in _workspaceHosts)
+            using (Dispatcher.CurrentDispatcher.DisableProcessing())
```

Comment:
When we're catching a host up, we flip the bit before calling this. If you just F12 on hostState.StartPushingToWorkspaceAndNotifyOfOpenDocuments() it should be checking the exact same bit on the exact same object. I believe you should be able to convince yourself of that statically.